src/main/java/seedu/address/logic/parser/DeleteFieldCommandParser.java

```diff
@@ -0,0 +1,145 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COMPANY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_JOBTITLE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRONOUN;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.DeleteFieldCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.EditPersonDescriptor;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Pronoun;
+import seedu.address.model.tag.Tag;
+
+public class DeleteFieldCommandParser implements Parser<DeleteFieldCommand> {
+    @Override
+    public DeleteFieldCommand parse(String userInput) throws ParseException {
+        throw new ParseException(MESSAGE_INVALID_VIEW);
+    }
+
+    /**
+     * Parses the user input to delete desired fields.
+     *
+     * @param args User input.
+     * @param personToDeleteField Person being edited.
+     * @return Command to delete fields of a person.
+     * @throws ParseException For invalid user input.
+     */
+    public DeleteFieldCommand parse(String args, Person personToDeleteField) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_COMPANY, PREFIX_JOBTITLE,
+                        PREFIX_PRONOUN, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+
+        if (!argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    DeleteFieldCommand.MESSAGE_USAGE));
+        }
+
+        EditPersonDescriptor deleteFieldDescriptor = new EditPersonDescriptor(personToDeleteField);
+
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            throw new ParseException(MESSAGE_INVALID_VIEW + "\n"
+                    + DeleteFieldCommand.MESSAGE_DELETE_NAME_FAILURE);
+        }
+
+        if (argMultimap.getValue(PREFIX_COMPANY).isPresent()) {
+            deleteFieldDescriptor.setCompany(null);
+        }
+
+        if (argMultimap.getValue(PREFIX_JOBTITLE).isPresent()) {
+            deleteFieldDescriptor.setJobTitle(null);
+        }
+
+        if (deleteFieldDescriptor.getNumbers().isPresent()) {
+            Collection<String> numbersToBeDeleted = argMultimap.getAllValues(PREFIX_PHONE);
+
+            assert numbersToBeDeleted != null;
```

Comment:
>  The raising of an exception indicates an unusual condition created by the user (e.g. user inputs an unacceptable input) or the environment (e.g., a file needed for the program is missing).
An assertion failure indicates the programmer made a mistake in the code (e.g., a null value is returned from a method that is not supposed to return null under any circumstances).

Should this be requireNonNull?